k2/csrc/properties.cc

```diff
@@ -50,4 +86,28 @@ bool IsEpsilonFree(const Fsa &fsa) {
   return true;
 }
 
+bool IsConnected(const Fsa &fsa) {
+  StateId num_states = fsa.NumStates();
+  std::vector<bool> accessible(num_states, false);
+  accessible[0] = true;  // the start state
+  for (const auto &arc : fsa.arcs) {
+    if (!accessible[arc.src_state]) return false;
+    accessible[arc.dest_state] = true;
+  }
+  if (std::find(accessible.begin(), accessible.end(), false) !=
+      accessible.end())
+    return false;
+
+  // reuse `accessible` to process co-accessible situation.
+  std::fill(accessible.begin(), accessible.end(), false);
+  accessible[num_states - 1] = true;  // the final state
+  int32_t num_arcs = fsa.arcs.size();
```

Comment:
static_cast, otherwise there will be a warning from the compiler.

Reply:
Sure, updated